fetch-in-react-practice/client-ts/src/components/Todos.tsx

```diff
@@ -0,0 +1,162 @@
+/* eslint-disable @typescript-eslint/no-unused-vars -- Remove me */
+import React, { useEffect, useState } from 'react';
+import PageTitle from './PageTitle';
+import TodoList from './TodoList';
+import TodoForm from './TodoForm';
+
+export type UnsavedTodo = {
+  task: string;
+  isCompleted: boolean;
+};
+export type Todo = UnsavedTodo & {
+  todoId: number;
+};
+
+export default function Todos() {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<unknown>();
+
+  /* Implement useEffect to fetch all todos. Hints are at the bottom of the file. */
+  useEffect(() => {
+    async function fetchTodos() {
+      try {
+        const res = await fetch('/api/todos');
+        if (!res.ok) {
+          throw new Error(`fetch Error ${res.status}`);
+        }
+        const todosList: Todo[] = await res.json();
+        setTodos(todosList);
+      } catch (err) {
+        setError(err);
+      } finally {
+        setIsLoading(false);
+      }
+    }
+    fetchTodos();
+  }, []);
+
+  /* Implement addTodo to add a new todo. Hints are at the bottom of the file. */
+  async function addTodo(newTodo: UnsavedTodo) {
```

Comment:
I think you should always explicitly state the return type of each of your functions. It makes the code more readable and robust since TS can make sure you are returning the data type you intend.

Since this is an async function, it has to return a Promise. Since you aren't explicitly returning anything, you should declare it as `: Promise<void>`